esmvaltool/diag_scripts/magic_bsc/combined_indices.r

```diff
@@ -34,317 +34,134 @@ model_names <- unique(unlist(unname(model_names)))
-  names(metadata)[1] <- var0
-  attr(data, "variables") <- metadata
-  variable_list <- list(variable = data, lat = lat, lon = lon, time = time)
-  names(variable_list)[1] <- var0
-
-  print(paste(
-    "Attribute projection from climatological data is saved and,",
-    "if it's correct, it can be added to the final output:",
-    projection)
-  )
+    data1 <- WeightedMean(data, lon = lon, lat = lat, #nolint
+                    region = c(lon_min[1], lon_max[1], lat_min[1], lat_max[1]),
+                    londim = 2, latdim = 3, mask = NULL)
+    data2 <- WeightedMean(data, lon = lon, lat = lat, #nolint
+                    region = c(lon_min[2], lon_max[2], lat_min[2], lat_max[2]),
+                    londim = 2, latdim = 3, mask = NULL)
+    data1 <- drop(data1)
+    data2 <- drop(data2)
+    data <- CombineIndices(list(data1, data2), weights = c(1, -1), #nolint
+                           operation = "add")
+}
 
-  dimlon <- ncdim_def(
-    name = "lon",
-    units = "degrees_east",
-    vals = as.vector(lon),
-    longname = "longitude"
-  )
-  dimlat <- ncdim_def(
-    name = "lat",
-    units = "degrees_north",
-    vals = as.vector(lat),
-    longname = "latitude")
-  dimtime <- ncdim_def(
-    name = "time",
-    units = "days since 1970-01-01 00:00:00",
-    vals = as.vector(as.numeric(time)),
-    longname = "time")
-  defdata <- ncvar_def(
-    name = "data",
-    units = units,
-    dim = list(time = dimtime),
-    longname = paste("Combination", long_names))
-    filencdf <- paste0(
-      plot_dir, "/", var0, "_", paste0(model_names, collapse = "_"),
-      "_", timestamp, "_", model_names_filename, "_", start_year, "_",
-      end_year, "_", ".nc")
-  file <- nc_create(filencdf, list(defdata))
-  ncvar_put(file, defdata, data)
-  nc_close(file)
+dimtime <- ncdim_def(name = "Time", units = "years",
+                     vals = starting : ending,
+                     longname = "Time")
+defdata <- ncvar_def(name = "data", units = units, dim = list(time = dimtime),
+               longname = paste("Index for region", region, "Variable", var0))
+filencdf <- paste0(work_dir, "/", var0, "_", timestamp, "_", months, "_",
+                   starting, ending, "_", ".nc")
+file <- nc_create(filencdf, list(defdata))
+ncvar_put(file, defdata, data)
+nc_close(file)
+
+
+png(paste0(plot_dir, "/", "Index_", region, ".png"), width = 7, height = 4,
+    units = "in", res = 150)
+plot(starting : ending, data, type = "l", col = "purple", lwd = 2, bty = "n",
+     xlab = "Time (years)", ylab = "Index",
+     main = paste("Region", region, "and Variable", var0))
+abline(h = 0, col = "grey", lty = 4)
+dev.off()
+
+
+# Set provenance for output files
+xprov <- list(ancestors = list(fullpath_filenames),
+              authors = list("hunt_al", "manu_ni"),
+              projects = list("c3s-magic"),
+              caption = "Combined selection",
+              statistics = list("other"),
+              moninf = params$moninf,
```

Comment:
Also here, there is no need to add the options specified in the recipe, esmvaltool already records those for you.